Sources/Fullscreen/MessageForm/MessageFormView.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import Foundation
+
+protocol MessageFormViewDelegate: AnyObject {
+    func messageFormView(_ view: MessageFormView, didEditMessageText text: String)
+}
+
+class MessageFormView: UIView {
+
+    // MARK: - UI properties
+
+    private lazy var textView: TextView = {
+        let textView = TextView(withAutoLayout: true)
+        textView.delegate = self
+        return textView
+    }()
+
+    private lazy var transparencyLabel: Label = {
+        let label = Label(style: .detail)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.numberOfLines = 0
+        label.text = viewModel.transparencyText
+        return label
+    }()
+
+    private lazy var transparencyLabelHeightConstraint = transparencyLabel.heightAnchor.constraint(equalToConstant: 0)
+
+    // MARK: - Internal properties
+
+    weak var delegate: MessageFormViewDelegate?
+
+    var text: String {
+        get {
+            return textView.text ?? ""
+        }
+        set {
+            textView.text = newValue
+            delegate?.messageFormView(self, didEditMessageText: newValue)
+        }
+    }
+
+    // MARK: - Private properties
+
+    private let viewModel: MessageFormViewModel
+
+    // MARK: - Init
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    required init(viewModel: MessageFormViewModel) {
+        self.viewModel = viewModel
+        super.init(frame: .zero)
+        setup()
+    }
+
+    private func setup() {
+        addSubview(textView)
+        addSubview(transparencyLabel)
+
+        // Adding a low-priority "infinite" height constraint to the TextView
+        // makes sure it gets chosen to fill the available vertical space.
+        let textViewHeightConstraint = textView.heightAnchor.constraint(lessThanOrEqualToConstant: 3000)
```

Comment:
Can you try content hugging etc instead?

Reply:
`label.setContentHuggingPriority(.defaultHigh, for: .vertical)` worked wonders. I must've done the layout for this class while sleeping, because I was using another constraint in addition to this to enforce the constraints.

Reply:
Looks a lot nicer with the new commit, good job! 😄